modules/pipeline/services/pipelinesvc/snippet.go

```diff
@@ -15,91 +15,63 @@ package pipelinesvc
-	batchSourceSnippetConfigs := make(map[string]map[string]apistructs.SnippetConfig)
-	singleSourceSnippetConfigs := make(map[string]map[string]apistructs.SnippetConfig)
-	for source, snippetConfigMap := range sourceSnippetConfigs {
-		// 若 source 支持批量查询，则批量查询
-		if _, supportAsync := sourceSupportAsyncBatch[source]; supportAsync {
-			batchSourceSnippetConfigs[source] = snippetConfigMap
-		} else { // 不支持，则单个查询
-			singleSourceSnippetConfigs[source] = snippetConfigMap
+	batchSourceSnippetConfigs := []apistructs.SnippetConfig{}
+	singleSourceSnippetConfigs := []apistructs.SnippetConfig{}
+	for _, snippetConfig := range sourceSnippetConfigs {
+		if _, supportAsync := sourceSupportAsyncBatch[snippetConfig.Source]; supportAsync {
+			batchSourceSnippetConfigs = append(batchSourceSnippetConfigs, snippetConfig)
+		} else {
+			singleSourceSnippetConfigs = append(singleSourceSnippetConfigs, snippetConfig)
 		}
 	}
 
-	// TODO 同一 source 考虑全局并发度，防止把客户端打满造成 timeout
-
-	yamlResults := make(map[string]map[string]string)
-
-	// 异步批量查询异步调用
-	var wg sync.WaitGroup
-	var batchErrs []error
-	var lock sync.Mutex
-	for i := range batchSourceSnippetConfigs {
-		wg.Add(1)
-
-		batchConfigs := batchSourceSnippetConfigs[i]
-		go func() {
-			defer wg.Done()
-
-			var snippetConfigs []apistructs.SnippetConfig
-			for _, cfg := range batchConfigs {
-				snippetConfigs = append(snippetConfigs, cfg)
-			}
-			batchResults, err := s.batchQueryPipelineYAMLBySnippetConfigs(snippetConfigs)
-			if err != nil {
-				lock.Lock()
-				batchErrs = append(batchErrs, err)
-				lock.Unlock()
-				return
-			}
-
-			lock.Lock()
-			for source, nameYamls := range batchResults {
-				for name, yml := range nameYamls {
-					if _, ok := yamlResults[source]; !ok {
-						yamlResults[source] = make(map[string]string)
-					}
-					yamlResults[source][name] = yml
-				}
-			}
-			lock.Unlock()
-		}()
+	var yamlResults = make(map[string]string)
+	batchResults, err := s.batchQueryPipelineYAMLBySnippetConfigs(batchSourceSnippetConfigs)
+	if err != nil {
+		return nil, err
 	}
-	wg.Wait()
-	if len(batchErrs) > 0 {
-		return nil, strutil.FlatErrors(batchErrs, ", ")
+	for k, v := range batchResults {
+		yamlResults[k] = v
 	}
 
-	// 不支持批量查询的仍单个调用
-	for i := range singleSourceSnippetConfigs {
-		singleConfigs := singleSourceSnippetConfigs[i]
-		for _, sc := range singleConfigs {
-			yml, err := s.queryPipelineYAMLBySnippetConfig(&sc)
-			if err != nil {
-				return nil, err
-			}
-			if _, ok := yamlResults[sc.Source]; !ok {
-				yamlResults[sc.Source] = make(map[string]string)
-			}
-			yamlResults[sc.Source][sc.Name] = yml
+	for _, singleConfig := range singleSourceSnippetConfigs {
+		yml, err := s.queryPipelineYAMLBySnippetConfig(&singleConfig)
+		if err != nil {
+			return nil, err
 		}
+		yamlResults[jsonparse.JsonOneLine(singleConfig)] = yml
+	}
+
+	var errMsgs []string
+	for _, snippetConfig := range sourceSnippetConfigs {
+		yml, ok := yamlResults[jsonparse.JsonOneLine(snippetConfig)]
```

Comment:
please test whether jsonparse.JsonOneLine will print the same result while config has map field.

if not same, please change map to character-ordered string slice and then json marshal.